src/test/java/io/jenkins/plugins/report/jtreg/main/comparator/JobsByQueryTest.java

```diff
@@ -0,0 +1,216 @@
+package io.jenkins.plugins.report.jtreg.main.comparator;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class JobsByQueryTest {
+    // WARNING! be really careful when changing this path, at the end of the test, the contents of this directory will be deleted!
+    private static final String pathToJobsDir = "src/test/resources/io/jenkins/plugins/report/jtreg/main/comparator/dummyJobs/";
+
+    @BeforeClass
+    public static void createDummyJobDirs() {
```

Comment:
Is it really necessary to have the palceholder? I think it will be better to move all those File to some tmp directory.
And even better would be the DirListing interface with in applciation provide  by "FsDirListing" ,built around parent dir, but in tests by "ListDirLISting" based upon simple ArrayList

Reply:
I have created the interface.